app/controllers/refusal_advice_controller.rb

```diff
@@ -0,0 +1,57 @@
+##
+# Controller to store advice, from the refusal advice wizard, given to users
+# after their Info Requests are marked as refused/rejected.
+#
+class RefusalAdviceController < ApplicationController
+  before_action :authenticate
+
+  def create
+    log_event
+
+    case parsed_refusal_advice_params.fetch(:button)
+    when 'clarification'
+      redirect_to new_request_followup_path(
+        request_id: info_request.id, anchor: 'followup'
+      )
+
+    when 'internal_review'
+      redirect_to new_request_followup_path(
+        request_id: info_request.id, internal_review: '1', anchor: 'followup'
+      )
+
+      # TODO: there are other actions - define what to do in the refusal advice
+      # YAML?
+    end
+  end
```

Comment:
Why not just do?

```ruby
    case parsed_refusal_advice_params.fetch(:button)
    when 'internal_review'
      redirect_to new_request_followup_path(
        request_id: info_request.id, internal_review: '1', anchor: 'followup'
      )
    else
      redirect_to new_request_followup_path(
        request_id: info_request.id, anchor: 'followup'
      )
    end
```

Reply:
Because all actions from the YAML get redirected through this controller.

Reply:
Currently, if a user submits an action which is not defined here, we get a `no_content` response:

```
No template found for RefusalAdviceController#create, rendering head :no_content
```

Reply:
YAML format has been updated to include the `target` as discussed in #6103